### ISSUE TITLE
Support < (less than) on Module, not just Class

### ISSUE THREAD

**Issue by jez** [sorbet/sorbet#1910]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Aclass%20A%0Aend%0A%0Asig%20%7Bparams(x%3A%20T.untyped).void%7D%0Adef%20foo(x)%0A%20%20if%20x.is_a%3F(Module)%20%26%26%20x%20%3C%20A%0A%20%20%20%20%20%20T.reveal_type(x)%20%23%20not%20ok...%3F%0A%20%20end%0A%20%20%0A%20%20if%20x.is_a%3F(Class)%20%26%26%20x%20%3C%20A%0A%20%20%20%20%20%20T.reveal_type(x)%20%23%20...ok%3F%0A%20%20end%0Aend)

```ruby
# typed: true
extend T::Sig

class A
end

sig {params(x: T.untyped).void}
def foo(x)
  if x.is_a?(Module) && x < A
      T.reveal_type(x) # not ok...?
  end
  
  if x.is_a?(Class) && x < A
      T.reveal_type(x) # ...ok?
  end
end
```

#### Observed output

```
editor.rb:10: Revealed type: Module https://srb.help/7014
    10 |      T.reveal_type(x) # not ok...?
              ^^^^^^^^^^^^^^^^
  From:
    editor.rb:8:
     8 |def foo(x)
                ^
    editor.rb:9:
     9 |  if x.is_a?(Module) && x < A
             ^^^^^^^^^^^^^^^^^^^^^^^^

editor.rb:14: Revealed type: T.class_of(A) https://srb.help/7014
    14 |      T.reveal_type(x) # ...ok?
              ^^^^^^^^^^^^^^^^
  From:
    editor.rb:8:
     8 |def foo(x)
                ^
    editor.rb:9:
     9 |  if x.is_a?(Module) && x < A
             ^^^^^^^^^^^^^^^^^^^^^^^^
    editor.rb:13:
    13 |  if x.is_a?(Class) && x < A
             ^^^^^^^^^^^^^^^^^^^^^^^
Errors: 2
```

#### Expected behavior

`Module` defines `Module#<`, which is a parent class of `Class`, so this code runs fine. We should reveal the same type in both branches.

- - -

If I had to venture a guess as to why we didn't implement this before, it's because it's a little bit tricky. In particular:

```ruby
    if x.is_a?(Class) && x <= M
      T.reveal_type(x)
    end
```

Basically, I think this might be dead code, but I can't say for sure. Any attempt to fix this issue should probably investigate what the behavior should be on this example, ideally either constructing an example that makes the body of this condition not dead or proving why it must be dead.

**Comment by DarkDimius** [comment]
```
[1] pry(main)> module M
[1] pry(main)* end
=> nil
[2] pry(main)> class C
[2] pry(main)*   include M
[2] pry(main)* end
=> C
[3] pry(main)> C.ancestors
=> [C, M, Object, JSON::Ext::Generator::GeneratorMethods::Object, PP::ObjectMixin, Kernel, BasicObject]
[4] pry(main)> C.is_a?(M.singleton_class)
=> false
```

Modules aren't inherited
e.g. if M has singleton method, you can't call it on C
We haven't implemented it because it would be wrong.

**Comment by jez** [comment]
Chatted IRL, we realized that `some module < some class` is actually fine, because when the comparison succeeds it implies that `some module` is actually a class (dmitry's note about `self.` methods applies for this case:

```ruby
    if x.is_a?(Class) && x <= M
      T.reveal_type(x)
    end
```

aka, `some class < some module`)

**Comment by jez** [comment]
Dupe of #887
